lib.php

```diff
@@ -114,9 +114,10 @@ function choicegroup_get_user_answer($choicegroup, $user, $returnArray = FALSE,
     } else {
         $user_answers = array();
     }
-
-    if (!count($choicegroup_groups)) {
-        $choicegroup_groups = choicegroup_get_groups($choicegroup);
+    if(is_array($choicegroup_groups)){
```

Comment:
shouldn't this be more like `if ( !is_array($choicegroup_groups) || !count($choicegroup_groups) )` ?

The code is trying to figure out whether the variable `$choicegroup_groups` is already set or not.

Reply:
It's just bad design originally (from me) to rely on global variables – I should probably rewrite this part properly.

Reply:
If i change the pull request adding is_array() .. would you be ok to merge this or you need to re-think the design ?

Reply:
I would be ok to merge